QuizApp/QuizApp/UI/Leaderboard/LeaderboardViewController.swift

```diff
@@ -0,0 +1,183 @@
+import Combine
+import UIKit
+
+class LeaderboardViewController: UIViewController {
+
+    private let gradient = CAGradientLayer()
+
+    private var viewModel: LeaderboardViewModel!
+    private var titleLabel: UILabel!
+    private var xButton: UIButton!
+    private var tableView: UITableView!
+    private var headerView: LeaderboardHeader!
+    private var cancellables = Set<AnyCancellable>()
+    private var quizId: Int!
+    private var leaderboard: [QuizLeaderboard] = []
+
+    init(viewModel: LeaderboardViewModel, quizId: Int) {
+        super.init(nibName: nil, bundle: nil)
+
+        self.viewModel = viewModel
+        self.quizId = quizId
```

Comment:
Why do you need to store quizId? If it is for starting a quiz, a VM should already have a reference to this ID so you can read it from there.